pkg/operator/ceph/file/mds/mds.go

```diff
@@ -119,84 +132,166 @@ func (c *Cluster) Start() error {
+			if err != nil {
+				return "", errors.Wrap(err, "failed to set default mds config options")
 			}
 		}
+	}
 
-		// start the deployment
-		d, err := c.makeDeployment(mdsConfig, c.fs.Namespace)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create deployment")
+	// start the deployment
+	d, err := c.makeDeployment(mdsConfig, c.fs.Namespace)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create deployment")
+	}
+	// Set owner ref to cephFilesystem object
+	err = c.ownerInfo.SetControllerReference(d)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to set owner reference for ceph filesystem %q secret", d.Name)
+	}
+
+	// Set the deployment hash as an annotation
+	err = patch.DefaultAnnotator.SetLastAppliedAnnotation(d)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to set annotation for deployment %q", d.Name)
+	}
+
+	_, createErr := c.context.Clientset.AppsV1().Deployments(c.fs.Namespace).Create(ctx, d, metav1.CreateOptions{})
+	if createErr != nil {
+		if !kerrors.IsAlreadyExists(createErr) {
+			return "", errors.Wrapf(createErr, "failed to create mds deployment %s", mdsConfig.ResourceName)
 		}
-		// Set owner ref to cephFilesystem object
-		err = c.ownerInfo.SetControllerReference(d)
+		logger.Infof("deployment for mds %s already exists. updating if needed", mdsConfig.ResourceName)
+		_, err = c.context.Clientset.AppsV1().Deployments(c.fs.Namespace).Get(ctx, d.Name, metav1.GetOptions{})
 		if err != nil {
-			return errors.Wrapf(err, "failed to set owner reference for mds deployment %q", d.Name)
+			return "", errors.Wrapf(err, "failed to get existing mds deployment %s for update", d.Name)
 		}
+	}
 
-		// Set the deployment hash as an annotation
-		err = patch.DefaultAnnotator.SetLastAppliedAnnotation(d)
-		if err != nil {
-			return errors.Wrapf(err, "failed to set annotation for deployment %q", d.Name)
+	if createErr != nil && kerrors.IsAlreadyExists(createErr) {
+		if err = UpdateDeploymentAndWait(c.context, c.clusterInfo, d, config.MdsType, daemonLetterID, c.clusterSpec.SkipUpgradeChecks, c.clusterSpec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
+			return "", errors.Wrapf(err, "failed to update mds deployment %s", d.Name)
 		}
+	}
+	return d.GetName(), nil
+}
 
-		_, createErr := c.context.Clientset.AppsV1().Deployments(c.fs.Namespace).Create(ctx, d, metav1.CreateOptions{})
-		if createErr != nil {
-			if !kerrors.IsAlreadyExists(createErr) {
-				return errors.Wrapf(createErr, "failed to create mds deployment %s", mdsConfig.ResourceName)
-			}
-			logger.Infof("deployment for mds %s already exists. updating if needed", mdsConfig.ResourceName)
-			_, err = c.context.Clientset.AppsV1().Deployments(c.fs.Namespace).Get(ctx, d.Name, metav1.GetOptions{})
-			if err != nil {
-				return errors.Wrapf(err, "failed to get existing mds deployment %s for update", d.Name)
-			}
-		}
+// isMajorCephMSUpgrade determine if mds version inferior than image
```

Comment:
Looks like a previous leftover from my suggestion since it applies to all upgrades instead of just major upgrades
```suggestion
// isCephUpgrade determine if mds version inferior than image
```